pkg/simulation/target.go

```diff
@@ -94,16 +94,19 @@ func (sim *Simulation) Neutrals() []key.TargetID {
 }
 
 func (sim *Simulation) Retarget(data info.Retarget) []key.TargetID {
-	// merged removal of filtered targets and targets in limbo into 1 loop.
-	i := 0
-	for _, target := range data.Targets {
-		if (data.IncludeLimbo || sim.Attr.HPRatio(target) > 0) && data.Filter(target) {
-			data.Targets[i] = target
-			i++
+	// filter Targets only if Filter is specified.
+	if data.Filter != nil {
+		// merged removal of filtered targets and targets in limbo into 1 loop.
+		i := 0
+		for _, target := range data.Targets {
+			if (data.IncludeLimbo || sim.Attr.HPRatio(target) > 0) && data.Filter(target) {
+				data.Targets[i] = target
+				i++
+			}
 		}
+		// truncate to safely remove unqualified targets.
+		data.Targets = data.Targets[:i]
 	}
```

Comment:
* We should still be filtering out limbo when `Filter` is `nil` (should be equivalent to having a filter that always returns `true`). 
* Add tests for having `Filter` unspecified and that it still filtering limbo is the expected behavior